code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm

```diff
@@ -0,0 +1,176 @@
+				to_chat(X, span_notice("We need [ally_plasma_cost - X.plasma_stored]u more blood to restore a sister."))
+				return FALSE
+			if(isdead(target))
+				to_chat(X, span_notice("We can only help living sisters."))
+				return FALSE
+			if(!do_mob(X, target, 1 SECONDS, BUSY_ICON_FRIENDLY, BUSY_ICON_MEDICAL))
+				return FALSE
+		return TRUE
+	to_chat(owner, span_notice("We can only drain or restore familiar biological lifeforms."))
+	return FALSE
+
+/datum/action/xeno_action/activable/rejuvenate/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	if(ishuman(A))
+		var/mob/living/carbon/human/target = A
+		while(X.health < X.maxHealth && do_after(X, 2 SECONDS, TRUE, A, BUSY_ICON_HOSTILE))
+			X.heal_wounds(1.1, TRUE)
+			X.adjust_sunder(-1.5)
+			target.blood_volume -= 2
+		to_chat(X, span_notice("We feel fully restored."))
+	else if(A == X)
+		X.use_plasma(self_plasma_cost)
+		X.apply_status_effect(/datum/status_effect/xeno_rejuvenate, self_rejuvenation_duration)
+		COOLDOWN_START(X, rejuvenate_self_cooldown, 20 SECONDS)
+		to_chat(X, span_notice("We tap into our reserves for nourishment."))
+	else
+		X.use_plasma(ally_plasma_cost)
+		var/mob/living/carbon/xenomorph/target_xeno = A
+		target_xeno.adjustBruteLoss(-target_xeno.maxHealth*0.3)
+		target_xeno.adjustFireLoss(-target_xeno.maxHealth*0.3)
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Carnage
+// ***************************************
+/datum/action/xeno_action/activable/carnage
+	name = "Carnage"
+	action_icon_state = "Carnage"
+	mechanics_text = "For a while your attacks drain blood and heal you. During Feast you also heal nearby allies."
+	ability_name = "carnage"
+	cooldown_timer = 40 SECONDS
+	plasma_cost = 0
+
+/datum/action/xeno_action/activable/carnage/use_ability()
+	var/mob/living/carbon/xenomorph/X = owner
+	X.apply_status_effect(/datum/status_effect/xeno_carnage, 20 SECONDS, X.xeno_caste.carnage_plasma_gain)
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Feast
+// ***************************************
+
+/datum/action/xeno_action/activable/feast
+	name = "Feast"
+	action_icon_state = "Feast"
+	mechanics_text = "Use a large amount of blood to get into a state of rejuvenation. During this time you use a small amount of blood and heal. You can cancel this early."
+	ability_name = "feast"
+	cooldown_timer = 180 SECONDS
+	plasma_cost = 100
+
+/datum/action/xeno_action/activable/feast/can_use_ability(atom/target, silent, override_flags)
+	. = ..()
+	var/mob/living/carbon/xenomorph/X = owner
+	if(X.has_status_effect(STATUS_EFFECT_XENO_FEAST))
+		return TRUE
+
+/datum/action/xeno_action/activable/feast/use_ability()
+	var/mob/living/carbon/xenomorph/X = owner
+	if(X.has_status_effect(STATUS_EFFECT_XENO_FEAST))
+		to_chat(X, span_notice("We decide to end our feast early..."))
+		X.remove_status_effect(STATUS_EFFECT_XENO_FEAST)
+		return
```

Comment:
cooldown is 180 seconds so you can only cancel 20 seconds early

and why would you want to cancel early anyway?

Reply:
the buff drains ~plasma~ stored blood over time and you can cancel to not waste it

Reply:
what tivi said is still an issue